### ISSUE TITLE
DB reformat

### ISSUE THREAD

**Issue by Sigi-cz** [modrzew/pokeminer#51]
what about switching to this DB format? seems much more logical, and should be more efficient, as now a lot of duplicate data are being stored (each sighting with position over and over again)
https://www.reddit.com/r/pokemongodev/comments/4ull2o/working_on_a_database_standard_for_map_objects/

**Comment by modrzew** [comment]
What? Duplicated data is not stored. There is a [duplicate check in place](https://github.com/modrzew/pokeminer/blob/master/db.py#L57).

I like the idea about using cell ID instead of lat/lon, I'll look into it as I have some time to spare.

**Comment by Sigi-cz** [comment]
i meant location of sighting being stored over and over as duplicate data.

**Comment by fu20161234** [comment]
I'm not familiar with cell ID, but I would personally vote to keep including lat/lon, and just adding this if desired. Also on the topic of DB, it might be helpful to put time spotted as well as time expires, since apparently some spawns are longer than 15 minutes.

**Comment by GaelicGamer** [comment]
Yeah we should keep the multiple records per location of sighting, its essentially the count at this point and helps to differentiate if there is a duplicate spawn of different time at the same location (its unlikely but it does happen), this is not duplicate data.

**Comment by Sigi-cz** [comment]
what are you talking about? did you even look at the DB concept?  
  
Now is the best time to switch thanks to the throttling, btw. Because you can only scan cell_ids where spawns are. (After you learn this info with first 1-2 hour scan)

**Comment by GaelicGamer** [comment]
Of course I read the schema, otherwise I would have no grounds to critique it with, thanks for assuming that I'm an idiot?

The poster of the schema even understood this and included a disclaimer in his post "**you may notice that there are no fields for frequently changing information like pokemon spawned or fort owner. the goal was to seperate static from "live" or mined data, best in a different db.**"

So lets say I want to get the spawns at location between location (x1, y2) and location (x2, y2) for determining if a nest exists in that location.

Presently this is a full index comparison over two indexed float values (I've changed the lat/lon to floats in my db but haven't suggested it to this repo just yet as I would rather re-write the entire schema once I have enough data to understand how spawns behave so I know what we should and shouldn't store) with no joins over a large set.  

If instead the schema you posted is implemented and we forgo maintaining the location with the pokemon_id as you suggest since it was your main point in the beginning we would instead have this:

For given (x1, y1) and (x2, y2) giving the rectangle of interest, I would now need to convert this into a set of s2 sphere ids including the area of interest and select them from the cells table, then join on the spawns from the spawn table, a full index scan over an indexed BIGINT, and then join on the live information for those spawns, which is another full index scan over an indexed BIGINT, to then be able to do these calculations or even check if I found a duplicate spawn.  I'm not sure how much SQL experience you have but considering the current sizes of live data for locations this would be foolish and would slow things down considerably.

We could implement this and also continue maintaining the location with the spawns in the live db, that is duplicate data but with a intended performance purpose. Maybe I'm misunderstanding you and you mean to say "We should implement this schema for determining the scanning points of where and when we need to scan, and then also overhaul the current live schema for storing the live data that we can then do analytics over."  But that is not what it sounded like in your original post or your follow up, and we all know that you scan using the cell_ids, which you can see in the code.

```
logger.info('Visiting point %d (%s %s)', i, point[0], point[1])
            self.api.set_position(point[0], point[1], 0)
            cell_ids = pgoapi_utils.get_cell_ids(point[0], point[1])
            self.api.set_position(point[0], point[1], 100)
            response_dict = self.api.get_map_objects(
                latitude=pgoapi_utils.f2i(point[0]),
                longitude=pgoapi_utils.f2i(point[1]),
                cell_id=cell_ids
            )
```

This is just my take on what I believe you are proposing from what you've said, and linked to thus far, and ultimately it is up to @modrzew but while I think there are some decent points to take away from this discussion, I don't believe you've done enough research to be making such passing remarks when others point out obvious flaws in what you suggest.

Again sorry for the slight hostility, but questioning if I've even read your suggested schema when I pointed out a direct flaw with it that even the schema owner seems to understand is very rude.
